### PR TITLE
Support JWT tokens for WebSocket APIs

### PR DESCRIPTION
**Pull request by Arshardh**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java [hunk 1]
```diff
@@ -192,6 +195,133 @@ public AuthenticationContext authenticate(String jwtToken, MessageContext synCtx
...
+     * access a particular API or not.
+     *
+     * @param jwtToken   The JWT token sent with the API request
+     * @param apiContext The context of the invoked API
+     * @param apiVersion The version of the invoked API
+     * @return an AuthenticationContext object which contains the authentication information
+     * @throws APISecurityException in case of authentication failure
+     */
+    @MethodStats
+    public AuthenticationContext authenticateForWebSocket(String jwtToken, String apiContext, String apiVersion)
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: We might have to add the revocation logic as well in here

Comment (Arshardh) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: We can add it once the revocation changes are merged

Comment (fazlan-nazeem) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: changes are merged https://github.com/wso2/carbon-apimgt/pull/7423

Comment (Arshardh) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]: added the revocation logic b861ac70300525979d3f146942ee522296f91a66

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java [hunk 2]
```diff
@@ -225,46 +229,113 @@ private boolean validateOAuthHeader(FullHttpRequest req)
...
+                            if (StringUtils.countMatches(apiKey, APIConstants.DOT) != 2) {
+                                log.debug("Invalid JWT token. The expected token format is <header.payload.signature>");
+                                throw new APISecurityException(APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
+                                        "Invalid JWT token");
+                            }
+                        }
+                    } catch (JSONException | IllegalArgumentException e) {
+                        isJwtToken = false;
+                        log.debug("Not a JWT token. Failed to decode the token header.", e);
                     }
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java]: You can use APIUtil.isvalidJWT method for this part

Comment (Arshardh) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java]: done

**Pull request merged**